pkg/sql/plan/query_builder.go

```diff
@@ -1202,6 +1202,9 @@ func (builder *QueryBuilder) buildTable(stmt tree.TableExpr, ctx *BindContext) (
 			TableDef:    tableDef,
 			BindingTags: []int32{builder.genNewTag()},
 		}, ctx)
+		if tableDef.TableType == "e" {
```

Comment:
you should set NodeType in buildTable。  not in remapAllColRefs。
because before remapAllColRefs we have some operator in TableScan Node(eg: pushdownFilters), that will make some unexpected problems.